### PR TITLE
Create a plugin for moving all plugins load up

### PR DESCRIPTION
**Pull request by rogeriochaves**
So, basically previously we were covering this scenario:

`B -> A!`

B depends on A which is a plugin. The way we did, B was waiting for A to load before executing its own code, this is fine, but we weren't covering that:

`C -> B -> A!`

B is still waiting for A to load asynchronously, but now C depends on B _synchronously_, and B returns undefined because well, it isn't ready yet. And then things break and burn.

This made me realize that a requirejs plugin actually stops the loading of _everything_ up the tree until it is resolved. So what I did was changing the loader to become a loader + a plugin, it works this way:
1. The loader annotates the js modules with dependencies to requirejs plugins with comments, so the webpack plugin can work on them later
2. The webpack plugin grabs the final resolved bundle from webpack, move the loading of the rjs plugins up, before the loading of the first module, and replaces the requires with the results of the loaders.

You can check the tests and the examples folder for more details

### CODE REVIEW

**Comment by rogeriochaves** [comment]
@pamo

File: test/parser.spec.js [hunk 1]
```diff
@@ -28,49 +28,44 @@ describe('parser', () => {
...
+          'use strict';
+          console.log('i18n', i18n);
+      });
     `));
   });
 
   describe('different AMD patterns', () => {
     it('parses anonymous modules', () => {
-      expect(parser.parse(moduleSyntaxFixtures.anonymousModule)).to.match(/examplePlugin\('args'/);
+      expect(parser.parse(moduleSyntaxFixtures.anonymousModule)).to.match(/example', \/\/ requirejs_plugin\|args\|/);
```
Comment (pamo) [inline test/parser.spec.js]: Can this pattern be extracted out into a variable?

**Comment by pamo** [comment]
Beleza

**Comment by pamo** [comment]
LGTM ⚡️

**Pull request merged**